atoma-proxy/src/server/middleware.rs

```diff
@@ -687,16 +687,35 @@ pub async fn handle_locked_stack_middleware(
             // We need to acquire a new stack for the request, to be able to retry
             let user_id = request_metadata.user_id;
             let max_total_num_compute_units = request_metadata.max_total_num_compute_units;
-            // 1. Acquire a new stack for the request, this will also lock compute units for the new acquired stack
-            let selected_node_metadata = get_cheapest_node_and_acquire_new_stack(
+            // 1. Try to get a Stack from the state manager
+            let maybe_stack = get_node_metadata_from_state_manager(
                 &state,
-                user_id,
                 &request_metadata.model_name,
-                &request_metadata.endpoint,
-                max_total_num_compute_units,
+                user_id,
+                max_total_num_compute_units as i64,
+                is_confidential_compute_endpoint(&endpoint),
```

Comment:
this is never confidential, because on line 677 you throw an error for confidential
so I just want to make sure, that you are aware of that

Reply:
It doesn't hurt to have the check in here too, and if we want to change the logic later on, it is more robust to have the check in place than if not